met-web/src/components/engagement/dashboard/report/EngagementDashboard.tsx

```diff
@@ -1,92 +1,124 @@
-import React, { useContext } from 'react';
-import { Grid } from '@mui/material';
-import { useAppSelector } from 'hooks';
-import { Link } from 'react-router-dom';
-import { useNavigate } from 'react-router';
-import { ActionContext } from '../../view/ActionContext';
-import { EngagementBanner } from '../../view/EngagementBanner';
-import { PreviewBanner } from '../../view/PreviewBanner';
-import { SubmissionStatus } from 'constants/engagementStatus';
+import React, { useEffect, useState } from 'react';
+import { Grid, Link as MuiLink, Skeleton, Typography } from '@mui/material';
+import { Link, useNavigate, useParams } from 'react-router-dom';
 import { AppConfig } from 'config';
-import { PrimaryButton } from 'components/common';
+import { MetPaper, PrimaryButton } from 'components/common';
+import { ReportBanner } from './ReportBanner';
+import { createDefaultEngagement, Engagement } from 'models/engagement';
+import { SubmissionStatus } from 'constants/engagementStatus';
+import { getEngagement } from 'services/engagementService';
+import { getErrorMessage } from 'utils';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { useAppDispatch } from 'hooks';
+
+export type EngagementParams = {
+    engagementId: string;
+};
 
 export const EngagementDashboard = () => {
-    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
-    const isPreview = isLoggedIn;
-    const { savedEngagement } = useContext(ActionContext);
+    const { engagementId } = useParams<EngagementParams>();
     const navigate = useNavigate();
+    const dispatch = useAppDispatch();
     const urlpath = AppConfig.redashDashboardUrl;
-    const isOpen = savedEngagement.submission_status === SubmissionStatus.Open;
 
-    const handleStartSurvey = () => {
-        if (!isPreview) {
-            return;
+    const [engagement, setEngagement] = useState<Engagement>(createDefaultEngagement());
+    const [isEngagementLoading, setEngagementLoading] = useState(true);
+
+    const validateEngagement = (engagementToValidate: Engagement) => {
+        const neverOpened = engagementToValidate?.submission_status < SubmissionStatus.Open;
+
+        if (neverOpened) {
+            throw new Error('Engagement has not yet been opened');
```

Comment:
what happens in the UII when we throw an error?

Reply:
This method is called under a try catch block. The error should be caught and this should run:

catch (error) {
                console.log(error);
                dispatch(
                    openNotification({
                        severity: 'error',
                        text: getErrorMessage(error) || 'Error occurred while fetching Engagement information',
                    }),
                );
                navigate('/');
            }

Reply:
hm..got it now.. 
I Would try to rewrite it a little bit ;but its alrite as it is..
Since the method is named as validateEngagement , exception raising is not  necessarily needed.. We are here using an exception to control the flow which is not ideal.. it can be a isValidEng or isOpenEngagement which returns a boolean and then u can do ur logic..isnt it more readable that way?
But definitely this is a well written code..so u ignore me for being picky

Reply:
It seems you prefer having an if statement to check an isValid boolean over handling validation via exception. This way is more straightforward sure, but I don't think it adheres better to the Open for extension Closed for change principle.

The try catch you can just extend the code if you needed another validation or check step.  You only have to write that method and call it inside the try.

But the boolean way, if you have to add another validation you will have to change your code to handle that boolean. Which violates the open close. I prefer the try catch and I actually think it's closer to the ideal